crates/build/src/docker.rs

```diff
@@ -0,0 +1,607 @@
+) -> Result<()> {
+    let client = Docker::connect_with_socket_defaults()?;
+
+    let mut entrypoint = vec!["/bin/bash".to_string(), "-c".to_string()];
+    let b_path_str = build_result_path
+        .as_os_str()
+        .to_str()
+        .context("Cannot convert Os String to String")?;
+    let mut cmds = vec![format!(
+        "mkdir -p target && cargo contract build {} --output-json > {}",
+        build_args, b_path_str
+    )];
+
+    entrypoint.append(&mut cmds);
+
+    let digest_code = container_digest(entrypoint.clone(), build_image.id.clone());
+    let container_name =
+        format!("ink-verified-{}-{}", contract_name, digest_code.clone());
+
+    let mut filters = HashMap::new();
+    filters.insert("name".to_string(), vec![container_name.clone()]);
+
+    let containers = client
+        .list_containers(Some(ListContainersOptions::<String> {
+            all: true,
+            filters,
+            ..Default::default()
+        }))
+        .await?;
+
+    let container_option = containers.first();
+
+    let mount = Mount {
+        target: Some(String::from(MOUNT_DIR)),
+        source: Some(
+            host_folder
+                .to_str()
+                .context("Cannot convert path to string.")?
+                .to_string(),
+        ),
+        typ: Some(MountTypeEnum::BIND),
+        ..Default::default()
+    };
+    let host_cfg = Some(HostConfig {
+        mounts: Some(vec![mount]),
+        ..Default::default()
+    });
+
+    let user;
+    #[cfg(unix)]
+    {
+        user = Some(format!(
+            "{}:{}",
+            users::get_current_uid(),
+            users::get_current_gid()
+        ));
+    };
+    #[cfg(windows)]
+    {
+        user = None;
+    }
+
+    let mut labels = HashMap::new();
+    labels.insert("digest-code".to_string(), digest_code);
+    let config = Config {
+        image: Some(build_image.id.clone()),
+        entrypoint: Some(entrypoint),
+        cmd: None,
+        labels: Some(labels),
+        host_config: host_cfg,
+        attach_stderr: Some(true),
+        tty: Some(true),
+        user,
+        ..Default::default()
+    };
+    let options = Some(CreateContainerOptions {
+        name: container_name.as_str(),
+        platform: Some("linux/amd64"),
+    });
+
+    let container_id = match container_option {
+        Some(container) => {
+            container
+                .id
+                .clone()
+                .context("Container does not have an ID")?
+        }
+        None => client.create_container(options, config).await?.id,
+    };
+
+    client
+        .start_container::<String>(&container_id, None)
+        .await?;
+
+    let start_time = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .context("Invalid start time")?
+        .as_secs() as i64;
+
+    maybe_println!(
+        verbosity,
+        " {} {}",
+        format!("{build_steps}").bold(),
+        "Started the build inside the container"
```

Comment:
It might be useful to print out the container id here, it would be useful for e.g. attaching to it. Or could add it as an `info!` at the very least.